rust/private/rustc.bzl

```diff
@@ -114,10 +116,20 @@ def collect_deps(deps, toolchain):
 
             # TODO: We could let the user choose how to link, instead of always preferring to link static libraries.
             libs = get_libs_for_static_executable(dep)
+            # TODO: This is changing the order of linker inputs, it shouldn't.
+            # In more sophisticated scenarios we need to retain the relative
+            # order of static/shared libraries. Imagine following libs:
+            #   libs = [ liba.a, libb.so, libc.a ]
+            # Ideally this will get converted into the following linker command
+            # line:
+            #   -la -lb -lc
+            # But the current implementation produces:
+            #    -la -lc -b
             dylibs = [l for l in libs.to_list() if l.basename.endswith(toolchain.dylib_ext)]
             staticlibs = [l for l in libs.to_list() if l.basename.endswith(toolchain.staticlib_ext)]
             transitive_dylibs = depset(transitive = [transitive_dylibs, depset(dylibs)])
             transitive_staticlibs = depset(transitive = [transitive_staticlibs, depset(staticlibs)])
+            linkopts.extend(dep[CcInfo].linking_context.user_link_flags)
```

Comment:
Doesn't this also need to be building up `transitive_linkopts` when rust -> rust -> c?